panic

```
panic: runtime error: index out of range [32] with length 32

goroutine 195 [running]:
github.com/startfellows/tongo/boc.(*BitString).On(...)
        /go/pkg/mod/github.com/startfellows/tongo@v0.11.2-0.20221222132608-35c544104384/boc/bitString.go:123
github.com/startfellows/tongo/boc.(*BitString).WriteBit(0x40?, 0xfe?)
        /go/pkg/mod/github.com/startfellows/tongo@v0.11.2-0.20221222132608-35c544104384/boc/bitString.go:307 +0x190
github.com/startfellows/tongo/boc.(*BitString).WriteBitString(0xc01ce98a80?, {{0xc01d63c680, 0x40, 0x40}, 0x1fe, 0x1fe, 0x101})
        /go/pkg/mod/github.com/startfellows/tongo@v0.11.2-0.20221222132608-35c544104384/boc/bitString.go:338 +0x105
github.com/startfellows/tongo/tlb.loadLabel(0xc032c42a80?, 0xc01ce98a80, 0x4?)
        /go/pkg/mod/github.com/startfellows/tongo@v0.11.2-0.20221222132608-35c544104384/tlb/hashmap.go:676 +0x385
github.com/startfellows/tongo/tlb.(*HashmapE[...]).mapInner(0xc01f144780, 0x100, 0x100, 0x28, 0x7f5605e595b8?)
        /go/pkg/mod/github.com/startfellows/tongo@v0.11.2-0.20221222132608-35c544104384/tlb/hashmap.go:344 +0x5f
github.com/startfellows/tongo/tlb.(*HashmapE[...]).UnmarshalTLB(0xc01f144780, 0xc01b342e08, {0xb12c5a, 0xc01f144780?})
        /go/pkg/mod/github.com/startfellows/tongo@v0.11.2-0.20221222132608-35c544104384/tlb/hashmap.go:334 +0x1b4
github.com/startfellows/tongo/tlb.decode(0xc01b342e08, {0xc3ec20?, 0xc008f4e410?, 0x3?}, {0xb12c5a, 0x7})
        /go/pkg/mod/github.com/startfellows/tongo@v0.11.2-0.20221222132608-35c544104384/tlb/decoder.go:36 +0x5ec
github.com/startfellows/tongo/tlb.decodeBasicStruct(0x0?, {0xba9ec0?, 0xc008f4e410?, 0x0?}, {0x0?, 0x0?})
        /go/pkg/mod/github.com/startfellows/tongo@v0.11.2-0.20221222132608-35c544104384/tlb/decoder.go:117 +0x1df
github.com/startfellows/tongo/tlb.decodeStruct(0xc009f88040?, {0xba9ec0?, 0xc008f4e410?, 0xbf7100?}, {0x0, 0x0})
        /go/pkg/mod/github.com/startfellows/tongo@v0.11.2-0.20221222132608-35c544104384/tlb/decoder.go:108 +0x125
github.com/startfellows/tongo/tlb.decode(0xc01b342e08, {0xba9ec0?, 0xc008f4e410?, 0xb246ab?}, {0x0, 0x0})
        /go/pkg/mod/github.com/startfellows/tongo@v0.11.2-0.20221222132608-35c544104384/tlb/decoder.go:97 +0x536
github.com/startfellows/tongo/tlb.decodeSumType(0xafc847?, {0xbf7100?, 0xc008f4e400?, 0x0?}, {0x0?, 0xb5c960?})
        /go/pkg/mod/github.com/startfellows/tongo@v0.11.2-0.20221222132608-35c544104384/tlb/decoder.go:140 +0x385
github.com/startfellows/tongo/tlb.decodeStruct(0xb3c1c0?, {0xbf7100?, 0xc008f4e400?, 0xc00c931be0?}, {0x0, 0x0})
        /go/pkg/mod/github.com/startfellows/tongo@v0.11.2-0.20221222132608-35c544104384/tlb/decoder.go:106 +0xdb
github.com/startfellows/tongo/tlb.decode(0xc01b342e08, {0xb3c1c0?, 0xc008f4e400?, 0xc00c931c50?}, {0x0, 0x0})
        /go/pkg/mod/github.com/startfellows/tongo@v0.11.2-0.20221222132608-35c544104384/tlb/decoder.go:97 +0x536
github.com/startfellows/tongo/tlb.Unmarshal(0xc00c931c88?, {0xb3c1c0?, 0xc008f4e400?})
        /go/pkg/mod/github.com/startfellows/tongo@v0.11.2-0.20221222132608-35c544104384/tlb/decoder.go:17 +0xaa
github.com/startfellows/tonapi/pkg/core.decodeFullContent(0xc823ca?, {0xc01d63c500, 0x40})
        /build-dir/pkg/core/nft.go:335 +0x6c


```

--- a/boc/bitString.go
+++ b/boc/bitString.go
@@ -47,7 +47,7 @@ func (s *BitString) Buffer() []byte {
 }
 
 func (s *BitString) checkRange(n int) error {
-	if n > s.cap {
+	if n >= s.cap {
 		return ErrBitStingOverflow
 	}
 	return nil

--- a/boc/bitString_test.go
+++ b/boc/bitString_test.go
@@ -13,3 +13,43 @@ func TestAppend(t *testing.T) {
 	s2.ReadBit()
 	s1.Append(s2)
 }
+
+func TestBitString_WriteBit(t *testing.T) {
+	bs1 := NewBitString(8)
+	for i := 0; i <= 7; i++ {
+		if err := bs1.WriteBit(true); err != nil {
+			t.Errorf("WriteBit() failed: %v", err)
+		}
+	}
+
+	tests := []struct {
+		name      string
+		bitstring BitString
+		value     bool
+		wantErr   string
+	}{
+		{
+			name:      "overflow",
+			bitstring: bs1,
+			value:     true,
+			wantErr:   "BitString overflow",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := tt.bitstring.WriteBit(tt.value)
+			if len(tt.wantErr) > 0 {
+				if err == nil {
+					t.Errorf("WriteBit() must return an error")
+				}
+				if err.Error() != tt.wantErr {
+					t.Errorf("WriteBit() error = %v, want = %v", err.Error(), tt.wantErr)
+				}
+				return
+			}
+			if err != nil {
+				t.Errorf("WriteBit() error = %v", err)
+			}
+		})
+	}
+}

